src/main/java/io/github/thebusybiscuit/slimefun4/core/networks/energy/EnergyNet.java

```diff
@@ -73,9 +72,9 @@ public static EnergyNet getNetworkFromLocationOrCreate(Location l) {
         }
     }
 
-    private final Set<Location> generators = new HashSet<>();
-    private final Set<Location> storage = new HashSet<>();
-    private final Set<Location> consumers = new HashSet<>();
+    private final Map<Location, EnergyNetComponent> generators = new HashMap<>();
+    private final Map<Location, EnergyNetComponent> capacitors = new HashMap<>();
+    private final Map<Location, EnergyNetComponent> consumers = new HashMap<>();
 
```

Comment:
Global variables... not at the top... NotLikeThis

Reply:
Not my code, I only refactored this. But I can change it